azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java

```diff
@@ -170,8 +170,13 @@ protected boolean isDeploymentDone(DeploymentResourceInner deploymentResource) {
         if (deploymentResource.properties().status() == DeploymentResourceStatus.PROCESSING) {
             return false;
         }
-        return !deploymentResource.properties().instances().stream()
-                .anyMatch(instance -> instance.status().equalsIgnoreCase("waiting") || instance.status().equalsIgnoreCase("pending"));
+        final String finalDiscoverStatus = deploymentResource.properties().active() ? "UP" : "OUT_OF_SERVICE";
+        final List<DeploymentInstance> instanceList = deploymentResource.properties().instances();
+        final boolean instanceStatus = !instanceList.stream().anyMatch(instance ->
+                instance.status().equalsIgnoreCase("waiting") || instance.status().equalsIgnoreCase("pending"));
+        final boolean discoverStatus = instanceList.stream().allMatch(instance ->
+                instance.discoveryStatus().equalsIgnoreCase(finalDiscoverStatus));
```

Comment:
reverse order also

Reply:
discoverStatus -> discoverStatusFinalized

Reply:
allMatch ? all instances should be OUT_OF_SERVICE?

Reply:
yes, the final status of inactive deployment instances shoudl be `OUT_OF_SERVICE`